usaspending_api/download/management/commands/download_sqs_worker.py

```diff
@@ -1,185 +1,154 @@
+            ``overwrite_error_message`` is True or if there is no prior error message on the record
+        overwrite_error_message (bool): if explicitly set to True, and error_message is not None, the provided error
+            message will overwrite an existing error message on the DownloadJob object. By default it will not.
+    """
+    download_job = _retrieve_download_job_from_db(download_job_id)
 
-def surrender_sqs_message_to_other_clients(message):
-    # https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html
-    set_sqs_message_visibility(message, 0)
+    download_job.job_status_id = JOB_STATUS_DICT[status]
+    if overwrite_error_message or (error_message and not download_job.error_message):
+        download_job.error_message = str(error_message) if error_message else None
 
+    download_job.save()
 
-def retrieve_download_job_from_db(download_job_id):
-    download_job = DownloadJob.objects.filter(download_job_id=download_job_id).first()
-    if download_job is None:
-        raise FatalError("Download Job {} record missing in DB!".format(download_job_id))
-    return download_job
 
+def _handle_queue_error(exc):
+    """Handles exceptions raised when processing a message from the queue.
 
-def update_download_record(download_job_id, status, error_message=None):
-    download_job = retrieve_download_job_from_db(download_job_id)
+    It handles two types:
+        QueueWorkerProcessError: an error in the job-processing code
+        QueueWorkDispatcherError: exception raised in the job-processing code
 
-    download_job.job_status_id = JOB_STATUS_DICT[status]
-    if error_message:
-        download_job.error_message = str(error_message)
+    Try to mark the job as failed, but continue raising the original Exception if not possible
+    This has likely already been set on the job from within error-handling code of the target of the
+    worker process, but this is here to for redundancy
+    """
+    try:
+        download_job_id = None
+        if exc.queue_message and exc.queue_message.body:
+            download_job_id = int(exc.queue_message.body)
+        log_job_message(
+            logger=logger,
+            message=f"{type(exc).__name__} caught. Attempting to fail the DownloadJob",
+            job_type=JOB_TYPE,
+            job_id=download_job_id,
+            is_exception=True,
+        )
+        if download_job_id:
+            stack_trace = "".join(traceback.format_exception(etype=type(exc), value=exc, tb=exc.__traceback__))
+            status = "failed"
+            _update_download_job_status(download_job_id, status, stack_trace)
+            log_job_message(
+                logger=logger,
+                message=f"Marked DownloadJob with id = {download_job_id} as {status}",
+                job_type=JOB_TYPE,
+                job_id=download_job_id,
+                is_exception=True,
+            )
+    except:  # noqa
```

Comment:
`except Exception:` is the recommended syntax, seems cleaner than a #noqa comment